libs/ts-rest/core/src/lib/dsl.ts

```diff
@@ -236,10 +237,11 @@ const recursivelyApplyOptions = <T extends AppRouter>(
               ? options.pathPrefix + value.path
               : value.path,
             headers: zodMerge(options?.baseHeaders, value.headers),
+            metadata: zodMerge(options?.metadata, value.metadata),
```

Comment:
We don't want to inadvertently set `metadata` to an empty object, if it is not set for in either router options or the route. Also no need to use `zodMerge` as we are not merging schemas.

```suggestion
            metadata: options?.metadata
              ? {
                  ...options?.metadata,
                  ...value.metadata,
                }
              : value.metadata,
```

Reply:
Got it. I didn't know about how the `zodMerge` function works. Also thought about having undefined if both are unset ;))

Reply:
Ahh, since `metadata` can be any type, it's unsafe to make an object and spread like this, isn't it...

Reply:
It's hard to tell when a route's metadata is a string and its router's metadata is an object. The resulting type would be an intersection between the two, but I'm not sure how it would behave in real-life scenarios...

<img width="1680" alt="image" src="https://github.com/ts-rest/ts-rest/assets/57426442/34a0c93a-8f84-46eb-b610-3e71bda60424">

Reply:
IMO, in most cases, the `metadata` should always be an object for consistency... 😃

Reply:
In that case, the only changes are like below, if you are comfortable 😃

<img width="680" alt="image" src="https://github.com/ts-rest/ts-rest/assets/57426442/5d29a275-3909-41a6-98c4-9b4c01d2fb00">

<img width="513" alt="image" src="https://github.com/ts-rest/ts-rest/assets/57426442/5706015c-876a-46e3-916e-54c2c45562aa">

Reply:
We can just let it be so it doesn't break any existing code. Nothing breaks as long as router-level metadata are used. It will only break if someone adds router-level metadata, and either of them are not objects, but it should be expected by the user that this breaks it.